lib/common/schemas.c

```diff
@@ -650,6 +652,51 @@ validate_xml(xmlNode *xml_blob, const char *validation, gboolean to_logs)
 }
 
 #if HAVE_LIBXSLT
+
+/* see crm_xml_error */
+static void
+crm_xslt_err(void *ctx, const char *msg, ...)
+G_GNUC_PRINTF(2, 3);
+
+static void
+crm_xslt_err(void *ctx, const char *msg, ...)
+{
+    int len = 0;
+    va_list args;
+    char *buf = NULL;
+    static int buffer_len = 0;
+    static char *buffer = NULL;
+
+    va_start(args, msg);
+    len = vasprintf(&buf, msg, args);
+
+    if (strchr(buf, '\n')) {
```

Comment:
That was plain copy-paste, really.
But that's not just that issue, it's broken more widely.
What if `vasprintf` fails?  Trap by trap.